spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -330,26 +333,44 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
     }
   }
 
-  def exprToProto(expr: Expression, input: Seq[Attribute]): Option[Expr] = {
+  def exprToProto(
+      expr: Expression,
+      input: Seq[Attribute],
+      binding: Boolean = true): Option[Expr] = {
     val conf = SQLConf.get
     val newExpr =
       DecimalPrecision.promote(conf.decimalOperationsAllowPrecisionLoss, expr, !conf.ansiEnabled)
-    exprToProtoInternal(newExpr, input)
+    exprToProtoInternal(newExpr, input, binding)
   }
 
-  def exprToProtoInternal(expr: Expression, inputs: Seq[Attribute]): Option[Expr] = {
+  /**
+   * Convert a Spark expression to protobuf.
+   *
+   * @param expr
+   *   The input expression
+   * @param inputs
+   *   The input attributes
+   * @param binding
+   *   Whether to bind the expression to the input attributes
+   * @return
+   *   The protobuf representation of the expression, or None if the expression is not supported
+   */
+  def exprToProtoInternal(
+      expr: Expression,
+      inputs: Seq[Attribute],
+      binding: Boolean): Option[Expr] = {
     SQLConf.get
     expr match {
       case a @ Alias(_, _) =>
-        exprToProtoInternal(a.child, inputs)
+        exprToProtoInternal(a.child, inputs, binding)
```

Comment:
Instead of changing all these, we can probably just move `exprToProtoInternal` and its friends into `exprToProto`. This also prevents it from being called accidentally outside.

Reply:
I don't get your point.

Reply:
You mean we remove `exprToProtoInternal`?

Reply:
Oh I mean move it inside `exprToProto`:

```scala
def exprToProto(...) {
  def exprToProtoInternal(...) { }
  ...
}
```

Reply:
Oh I see. You don't want `exprToProtoInternal` to be called outside?

Reply:
Right, I think it shouldn't be called directly except by `exprToProto` or itself.

Reply:
Okay